From 6cdd3f70b84f11fbc42861700b7f77547e771bc0 Mon Sep 17 00:00:00 2001
From: Nick Dimiduk <ndimiduk@apache.org>
Date: Thu, 18 Sep 2014 17:48:20 -0400
Subject: [PATCH] HBASE-11178 Remove deprecation annotations from mapred
 namespace

Our parent project is not dropping its legacy API, so neither shall we. Remove
the deprecation annotations from that implementation.
---
 hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/Driver.java   | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/GroupingTableMap.java  | 1 -
 .../main/java/org/apache/hadoop/hbase/mapred/HRegionPartitioner.java    | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableMap.java  | 1 -
 .../main/java/org/apache/hadoop/hbase/mapred/IdentityTableReduce.java   | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/RowCounter.java        | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormat.java  | 1 -
 .../main/java/org/apache/hadoop/hbase/mapred/TableInputFormatBase.java  | 1 -
 hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMap.java | 1 -
 .../main/java/org/apache/hadoop/hbase/mapred/TableMapReduceUtil.java    | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/TableOutputFormat.java | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReader.java | 1 -
 .../main/java/org/apache/hadoop/hbase/mapred/TableRecordReaderImpl.java | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/TableReduce.java       | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/TableSplit.java        | 1 -
 .../src/main/java/org/apache/hadoop/hbase/mapred/package-info.java      | 2 +-
 .../src/main/java/org/apache/hadoop/hbase/mapreduce/package-info.java   | 2 +-
 17 files changed, 2 insertions(+), 17 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/Driver.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/Driver.java
index c13b489..6ce4492 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/Driver.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/Driver.java
@@ -27,7 +27,6 @@ import com.google.common.annotations.VisibleForTesting;
  * Driver for hbase mapreduce jobs. Select which to run by passing name of job
  * to this main.
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class Driver {
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/GroupingTableMap.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/GroupingTableMap.java
index 9330d6c..6cd0602 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/GroupingTableMap.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/GroupingTableMap.java
@@ -38,7 +38,6 @@ import org.apache.hadoop.mapred.Reporter;
 /**
  * Extract grouping columns from input record
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class GroupingTableMap
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/HRegionPartitioner.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/HRegionPartitioner.java
index 55010ba..60a2c99 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/HRegionPartitioner.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/HRegionPartitioner.java
@@ -41,7 +41,6 @@ import org.apache.hadoop.mapred.Partitioner;
  * @param <K2>
  * @param <V2>
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class HRegionPartitioner<K2,V2>
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableMap.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableMap.java
index 5b793ab..2f5a9b8 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableMap.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableMap.java
@@ -32,7 +32,6 @@ import org.apache.hadoop.mapred.Reporter;
 /**
  * Pass the given key and record as-is to reduce
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class IdentityTableMap
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableReduce.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableReduce.java
index e101875..8d4d301 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableReduce.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/IdentityTableReduce.java
@@ -34,7 +34,6 @@ import org.apache.hadoop.mapred.Reporter;
 /**
  * Write to table each key, record pair
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class IdentityTableReduce
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/RowCounter.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/RowCounter.java
index b576151..b6f26b4 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/RowCounter.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/RowCounter.java
@@ -41,7 +41,6 @@ import org.apache.hadoop.util.ToolRunner;
  * Map outputs table rows IF the input row has columns that have content.
  * Uses an {@link IdentityReducer}
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class RowCounter extends Configured implements Tool {
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormat.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormat.java
index ff462bb..6c96f86 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormat.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormat.java
@@ -36,7 +36,6 @@ import org.apache.hadoop.util.StringUtils;
 /**
  * Convert HBase tabular data into a format that is consumable by Map/Reduce.
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class TableInputFormat extends TableInputFormatBase implements
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormatBase.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormatBase.java
index e08565f..98debee 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormatBase.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableInputFormatBase.java
@@ -65,7 +65,6 @@ import org.apache.hadoop.mapred.Reporter;
  * </pre>
  */
 
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public abstract class TableInputFormatBase
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMap.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMap.java
index de12f38..3a57bc3 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMap.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMap.java
@@ -32,7 +32,6 @@ import org.apache.hadoop.mapred.Mapper;
  * @param <K> WritableComparable key class
  * @param <V> Writable value class
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public interface TableMap<K extends WritableComparable<? super K>, V>
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMapReduceUtil.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMapReduceUtil.java
index f662e42..0f03159 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMapReduceUtil.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableMapReduceUtil.java
@@ -49,7 +49,6 @@ import org.apache.zookeeper.KeeperException;
 /**
  * Utility for {@link TableMap} and {@link TableReduce}
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 @SuppressWarnings({ "rawtypes", "unchecked" })
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableOutputFormat.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableOutputFormat.java
index b2e2698..0f7cded 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableOutputFormat.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableOutputFormat.java
@@ -41,7 +41,6 @@ import org.apache.hadoop.util.Progressable;
 /**
  * Convert Map/Reduce output and write it to an HBase table
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class TableOutputFormat extends
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReader.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReader.java
index 1f201a7..2e64649 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReader.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReader.java
@@ -33,7 +33,6 @@ import org.apache.hadoop.mapred.RecordReader;
 /**
  * Iterate over an HBase table data, return (Text, RowResult) pairs
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class TableRecordReader
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReaderImpl.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReaderImpl.java
index a574a0b..fc5581c 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReaderImpl.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableRecordReaderImpl.java
@@ -42,7 +42,6 @@ import static org.apache.hadoop.hbase.mapreduce.TableRecordReaderImpl.LOG_PER_RO
 /**
  * Iterate over an HBase table data, return (Text, RowResult) pairs
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class TableRecordReaderImpl {
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableReduce.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableReduce.java
index af1bfd7..2ca7cf8 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableReduce.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableReduce.java
@@ -31,7 +31,6 @@ import org.apache.hadoop.mapred.Reducer;
  * @param <K> key class
  * @param <V> value class
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 @SuppressWarnings("unchecked")
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableSplit.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableSplit.java
index 1b8d2a9..237fe47 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableSplit.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/TableSplit.java
@@ -33,7 +33,6 @@ import org.apache.hadoop.mapred.InputSplit;
 /**
  * A table split corresponds to a key range [low, high)
  */
-@Deprecated
 @InterfaceAudience.Public
 @InterfaceStability.Stable
 public class TableSplit implements InputSplit, Comparable<TableSplit> {
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/package-info.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/package-info.java
index 8ff35a3..8a2a363 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/package-info.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapred/package-info.java
@@ -18,7 +18,7 @@
  */
 /**
 Provides HBase <a href="http://wiki.apache.org/hadoop/HadoopMapReduce">MapReduce</a>
-Input/OutputFormats, a table indexing MapReduce job, and utility
+Input/OutputFormats, a table indexing MapReduce job, and utility methods.
 
 <p>See <a href="http://hbase.apache.org/book.html#mapreduce">HBase and MapReduce</a>
 in the HBase Reference Guide for mapreduce over hbase documentation. 
diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/package-info.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/package-info.java
index f811e21..199e168 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/package-info.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mapreduce/package-info.java
@@ -18,7 +18,7 @@
  */
 /**
 Provides HBase <a href="http://wiki.apache.org/hadoop/HadoopMapReduce">MapReduce</a>
-Input/OutputFormats, a table indexing MapReduce job, and utility
+Input/OutputFormats, a table indexing MapReduce job, and utility methods.
 
 <p>See <a href="http://hbase.apache.org/book.html#mapreduce">HBase and MapReduce</a>
 in the HBase Reference Guide for mapreduce over hbase documentation. 
-- 
1.9.0

